src/display.cpp

```diff
@@ -89,29 +89,27 @@ namespace {
 unsigned int display::zoom_ = DefaultZoom;
 unsigned int display::last_zoom_ = SmallZoom;
 
-// Factored out of set_zoom() so it can be re-used in constructor (loading from preferences)
-// new_zoom is the zoom factor (corresponding to display::zoom_)
-// new_zoom_index is the index of zoom_levels corresponding to zoom (corresponding to display::zoom_index_)
-// Returns true when validated and and zoom_index set and false if zoom cannot be validated (in which case parameters remain unchanged)
-static bool validate_zoom_level_and_set_index(unsigned int& new_zoom, int& new_zoom_index)
+// Returns index of zoom_levels which is closest match to input tile_size
+// Assumption: zoom_levels is a sorted vector of ascending tile sizes
+// (game_config is set-up this way as of October 2019 but there is no documentation stating it has to be so)
+int get_zoom_levels_index(unsigned int zoom_level)
 {
-	auto iter = std::lower_bound(zoom_levels.begin(), zoom_levels.end(), new_zoom);
+	zoom_level = utils::clamp(zoom_level, MinZoom, MaxZoom);	// ensure zoom_level is within zoom_levels bounds
+	auto iter = std::lower_bound(zoom_levels.begin(), zoom_levels.end(), zoom_level);
 
-	if(iter == zoom_levels.end()) {
-		return false;
-	} else if(iter != zoom_levels.begin()) {
+	// find closest match
+	if(iter != zoom_levels.begin() && iter != zoom_levels.end()) {
```

Comment:
`iter != zoom_levels.end()` is always true because of the clamp, isn't it?

Reply:
Yes, that's the 'this should never happen' comment in the old code. But given how many times this code has been chopped and changed I thought having the condition will still at least make it clear that we are somewhere in between the beginning and end of the vector. I can remove it if you really wish.

Reply:
No, it's fine, I was just asking to make sure it really was a "can't happen" thing and not something else.